### DISCUSSION TITLE
*csv table overwrite dynamically

### DISCUSSION THREAD

**Discussion by h3tz** [getgauge/gauge#2225] [General]
Hi there. 

we are using the table.csv on specification. In our situation, we would like to create the *.csv within in gauge before it will be used. We tried the before hooks but it seems like the table: xxx in the specification statement in the *.spec will be executed before before hooks.
Is there something we are doing wrong or any other solution for example to overwrite the existing *.csv table anytime?
Many thanks.

**Comment by zabil** [comment]
> Is there something we are doing wrong or any other solution for example to overwrite the existing *.csv table anytime?

Gauge parses and validates the csv header with the spec before executing it. For this reason it's not possible to generate a csv during execution. The recommended approach is to generate the csv (as a pre-test script for example) and then run Gauge.

**Comment by h3tz** [comment]
Yes, that's the approach we are currently doing. For any kind of test automation that's not an issue. For "manual" execution it's not straightforward for most of our customers.
Many Thanks.
